CMakeLists.txt

```diff
@@ -21,6 +21,7 @@ project(libwpe VERSION "${PROJECT_VERSION}")
 
 set(WPE_BACKEND "" CACHE STRING
     "Name of the backend library to load, instead of libWPEBackend-default.so")
+set(WPE_DISABLE_XKB_INPUT OFF CACHE STRING "Disable use of libxkbcommon for keyboard input")
```

Comment:
We chatted about this directly but for posterity I'll note it here.

The reason I went with `DISABLE` vs `ENABLE` was because an `ENABLE` would break clients of the code since this bit of code was always enabled before. So they all would need to add that define somehow. I'm not familiar enough with the *nix ecosystem to know whether or not this would be a problem when something like `pkg-config` is used or if theres something in Meson similar to CMake's project exports that would also prevent breakage.

Happy to make the change if its what you want to do.

Reply:
In our chat you made the good point that turning into an `ENABLE` and not putting guards in the headers (which was my suggestion) would cause a failure at link time for libwpe builds with XKB disabled — and that it would be desirable for the failure to pop earlier at build time.

Nevertheless, using `DISABLE` (as currently in the PR) will not cause a failure at build time when either if one forgets to pass `-DWPE_DISABLE_XKB_INPUT=1` while compiling code that uses libwpe, either.

So, I would still go with the suggested `ENABLE` version and not using guards in the headers, or using guards of this form:

```c
#if !defined(WPE_ENABLE_XKB_INPUT) || WPE_ENABLE_XKB_INPUT
/* code that uses XKB */
#endif
```

This way when the macro is undefined, we still get all the definitions. If we want to have build errors crop up earlier, then we need to add the `-DWPE_ENABLE_XKB_INPUT=1|0` in the `Cflags:` field of the generated `wpe.pc` pkg-config module file (and as explained above this would have been needed also for the negative form of the option!)